### PR TITLE
`Plagiarism detection`: Fix plagiarism comparisons pagination

### PR DESCRIPTION
**Pull request by jakubriegel**
#### General
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
Resolves part of #6593 

### Description
- fixes pagination numbering: when server returned 100 plagiarism comparisons the client created two pages in the sidebar, despite the second one was empty; now inly one page is created
- introduces a warning that if there are more than 100 plagiarism comparisons only top 100 of them are returned

### Steps for Testing

**Pagination numbering**
Change is simple and is verified by unit test.

**Trimmed comparisons warning**
Prerequisites:
- 1 Instructor
- 1 exercise with plagiarism detected
- local set up

1.  Remove the `*ngIf` in  `plagiarism-sidebar.component.html:11`.
2. Go to plagiarism comparisons page.
3. Verify that the warning is visible.

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| plagiarism-sidebar.component.ts | 97% | ✅                           |

### Screenshots
Warning in EN:
image

Warning in DE:
image

### CODE REVIEW

File: src/main/webapp/app/exercises/shared/plagiarism/plagiarism-sidebar/plagiarism-sidebar.component.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export class PlagiarismSidebarComponent implements OnChanges {
     }
 
     computeNumberOfPages(totalComparisons: number) {
```
Comment (lennart-keller) [inline src/main/webapp/app/exercises/shared/plagiarism/plagiarism-sidebar/plagiarism-sidebar.component.ts:72]: Maybe you can rename it to something like maximumPageIndex (and currentPageIndex in other places in the code) or so?

File: src/main/webapp/app/exercises/shared/plagiarism/plagiarism-sidebar/plagiarism-sidebar.component.html [hunk 2]
```diff
@@ -7,6 +7,11 @@
         <fa-icon class="text-warning disclaimer-icon" [icon]="faExclamationTriangle"></fa-icon>
         <span>{{ 'artemisApp.plagiarism.casesFiltered' | artemisTranslate }}</span>
     </div>
+    <!--  Temporary solution until #6593 is fixed  -->
+    <div class="filter-disclaimer" *ngIf="comparisons?.length === 1">
```
Comment (jakubriegel) [inline src/main/webapp/app/exercises/shared/plagiarism/plagiarism-sidebar/plagiarism-sidebar.component.html:11]: `length === 100`

**Comment by jakubriegel** [comment]
After testing session:
- refactor number of pages so that it is not 0 when there is single page
- fix `length ===`

**Review by pal03377 — APPROVED** [review]
Got a local demonstration in the testing session. Code also looks good 👍

**Review by b-fein — APPROVED** [review]
Thanks! Code looks good.

**Review by Strohgelaender — APPROVED** [review]
code lgtm

**Review by lennart-keller — APPROVED** [review]
Saw demonstration in testing session. Code lgtm, as well.

**Pull request merged**